### ISSUE TITLE
rpiCameraHFlip and rpiCameraVFlip don't flip the view?

### ISSUE THREAD

**Issue by yllekz** [aler9/rtsp-simple-server#1257]
## Which version are you using?

Latest Docker release

## Which operating system are you using?

- [ ] Linux amd64 standard
- [ ] Linux amd64 Docker
- [ ] Linux arm64 standard
- [ ] Linux arm64 Docker
- [ ] Linux arm7 standard
- [X] Linux arm7 Docker [Raspberry Pi 4B - 64 bit Raspberry Pi OS]
- [ ] Linux arm6 standard
- [ ] Linux arm6 Docker
- [ ] Windows amd64 standard
- [ ] Windows amd64 Docker (WSL backend)
- [ ] macOS amd64 standard
- [ ] macOS amd64 Docker
- [ ] Other (please describe)

## Describe the issue

I see there were parameters added to the config file to flip the camera view horizontally and vertically, however when adjusting these values and restarting everything, the camera orientation still does not change.

## Describe how to replicate the issue

1. adjust rtsp-simple-server.yml file's ``rpiCameraHFlip`` value to ``true`` (Same for ``rpiCameraVFlip`` for testing purposes)
2. restart containers/server (tried both)
3. Observe that the camera still isn't rotated in any way

## Did you attach the server logs?

Basically nothing in the Logs that assist with this.

## Did you attach a network dump?

no

**Comment by aler9** [comment]
Sorry but i wasn't able to replicate the issue, both parameters work for me. Can you post a sample configuration file?

**Comment by yllekz** [comment]
Sure, here is my config file:

```
###############################################
# General parameters

# Sets the verbosity of the program; available values are "error", "warn", "info", "debug".
logLevel: info
# Destinations of log messages; available values are "stdout", "file" and "syslog".
logDestinations: [stdout]
# If "file" is in logDestinations, this is the file which will receive the logs.
logFile: rtsp-simple-server.log

# Timeout of read operations.
readTimeout: 10s
# Timeout of write operations.
writeTimeout: 10s
# Number of read buffers.
# A higher value allows a wider throughput, a lower value allows to save RAM.
readBufferCount: 512

# HTTP URL to perform external authentication.
# Every time a user wants to authenticate, the server calls this URL
# with the POST method and a body containing:
# {
#   "ip": "ip",
#   "user": "user",
#   "password": "password",
#   "path": "path",
#   "action": "read|publish"
#   "query": "url's raw query"
# }
# If the response code is 20x, authentication is accepted, otherwise
# it is discarded.
externalAuthenticationURL:

# Enable the HTTP API.
api: no
# Address of the API listener.
apiAddress: 127.0.0.1:9997

# Enable Prometheus-compatible metrics.
metrics: no
# Address of the metrics listener.
metricsAddress: 127.0.0.1:9998

# Enable pprof-compatible endpoint to monitor performances.
pprof: no
# Address of the pprof listener.
pprofAddress: 127.0.0.1:9999

# Command to run when a client connects to the server.
# This is terminated with SIGINT when a client disconnects from the server.
# The following environment variables are available:
# * RTSP_PORT: server port
runOnConnect:
# Restart the command if it exits suddenly.
runOnConnectRestart: no

###############################################
# RTSP parameters

# Disable support for the RTSP protocol.
rtspDisable: no
# List of enabled RTSP transport protocols.
# UDP is the most performant, but doesn't work when there's a NAT/firewall between
# server and clients, and doesn't support encryption.
# UDP-multicast allows to save bandwidth when clients are all in the same LAN.
# TCP is the most versatile, and does support encryption.
# The handshake is always performed with TCP.
protocols: [udp, multicast, tcp]
# Encrypt handshakes and TCP streams with TLS (RTSPS).
# Available values are "no", "strict", "optional".
encryption: "no"
# Address of the TCP/RTSP listener. This is needed only when encryption is "no" or "optional".
rtspAddress: :8554
# Address of the TCP/TLS/RTSPS listener. This is needed only when encryption is "strict" or "optional".
rtspsAddress: :8322
# Address of the UDP/RTP listener. This is needed only when "udp" is in protocols.
rtpAddress: :8000
# Address of the UDP/RTCP listener. This is needed only when "udp" is in protocols.
rtcpAddress: :8001
# IP range of all UDP-multicast listeners. This is needed only when "multicast" is in protocols.
multicastIPRange: 224.1.0.0/16
# Port of all UDP-multicast/RTP listeners. This is needed only when "multicast" is in protocols.
multicastRTPPort: 8002
# Port of all UDP-multicast/RTCP listeners. This is needed only when "multicast" is in protocols.
multicastRTCPPort: 8003
# Path to the server key. This is needed only when encryption is "strict" or "optional".
# This can be generated with:
# openssl genrsa -out server.key 2048
# openssl req -new -x509 -sha256 -key server.key -out server.crt -days 3650
serverKey: server.key
# Path to the server certificate. This is needed only when encryption is "strict" or "optional".
serverCert: server.crt
# Authentication methods.
authMethods: [basic, digest]

###############################################
# RTMP parameters

# Disable support for the RTMP protocol.
rtmpDisable: no
# Address of the RTMP listener. This is needed only when encryption is "no" or "optional".
rtmpAddress: :1935
# Encrypt connections with TLS (RTMPS).
# Available values are "no", "strict", "optional".
rtmpEncryption: "no"
# Address of the RTMPS listener. This is needed only when encryption is "strict" or "optional".
rtmpsAddress: :1936
# Path to the server key. This is needed only when encryption is "strict" or "optional".
# This can be generated with:
# openssl genrsa -out server.key 2048
# openssl req -new -x509 -sha256 -key server.key -out server.crt -days 3650
rtmpServerKey: server.key
# Path to the server certificate. This is needed only when encryption is "strict" or "optional".
rtmpServerCert: server.crt

###############################################
# HLS parameters

# Disable support for the HLS protocol.
hlsDisable: no
# Address of the HLS listener.
hlsAddress: :8888
# By default, HLS is generated only when requested by a user.
# This option allows to generate it always, avoiding the delay between request and generation.
hlsAlwaysRemux: no
# Variant of the HLS protocol to use. Available options are:
# * mpegts - uses MPEG-TS segments, for maximum compatibility.
# * fmp4 - uses fragmented MP4 segments, more efficient.
# * lowLatency - uses Low-Latency HLS.
hlsVariant: mpegts
# Number of HLS segments to keep on the server.
# Segments allow to seek through the stream.
# Their number doesn't influence latency.
hlsSegmentCount: 7
# Minimum duration of each segment.
# A player usually puts 3 segments in a buffer before reproducing the stream.
# The final segment duration is also influenced by the interval between IDR frames,
# since the server changes the duration in order to include at least one IDR frame
# in each segment.
hlsSegmentDuration: 1s
# Minimum duration of each part.
# A player usually puts 3 parts in a buffer before reproducing the stream.
# Parts are used in Low-Latency HLS in place of segments.
# Part duration is influenced by the distance between video/audio samples
# and is adjusted in order to produce segments with a similar duration.
hlsPartDuration: 200ms
# Maximum size of each segment.
# This prevents RAM exhaustion (Default 50M)
hlsSegmentMaxSize: 50M
# Value of the Access-Control-Allow-Origin header provided in every HTTP response.
# This allows to play the HLS stream from an external website.
hlsAllowOrigin: '*'
# Enable TLS/HTTPS on the HLS server.
# This is required for Low-Latency HLS.
hlsEncryption: no
# Path to the server key. This is needed only when encryption is yes.
# This can be generated with:
# openssl genrsa -out server.key 2048
# openssl req -new -x509 -sha256 -key server.key -out server.crt -days 3650
hlsServerKey: server.key
# Path to the server certificate.
hlsServerCert: server.crt
# List of IPs or CIDRs of proxies behind the HLS server.
# If the server receives a request from one of these entries, IP in logs
# will be taken from the X-Forwarded-For header.
hlsTrustedProxies: []

###############################################
# Path parameters

# These settings are path-dependent, and the map key is the name of the path.
# It's possible to use regular expressions by using a tilde as prefix.
# For example, "~^(test1|test2)$" will match both "test1" and "test2".
# For example, "~^prefix" will match all paths that start with "prefix".
# The settings under the path "all" are applied to all paths that do not match
# another entry.
paths:
  all:
    # Source of the stream. This can be:
    # * publisher -> the stream is published by a RTSP or RTMP client
    # * rtsp://existing-url -> the stream is pulled from another RTSP server / camera
    # * rtsps://existing-url -> the stream is pulled from another RTSP server / camera with RTSPS
    # * rtmp://existing-url -> the stream is pulled from another RTMP server / camera
    # * rtmps://existing-url -> the stream is pulled from another RTMP server / camera with RTMPS
    # * http://existing-url/stream.m3u8 -> the stream is pulled from another HLS server
    # * https://existing-url/stream.m3u8 -> the stream is pulled from another HLS server with HTTPS
    # * redirect -> the stream is provided by another path or server
    # * rpiCamera -> the stream is provided by a Raspberry Pi Camera
    source: publisher

    # If the source is an RTSP or RTSPS URL, this is the protocol that will be used to
    # pull the stream. available values are "automatic", "udp", "multicast", "tcp".
    sourceProtocol: automatic

    # Tf the source is an RTSP or RTSPS URL, this allows to support sources that
    # don't provide server ports or use random server ports. This is a security issue
    # and must be used only when interacting with sources that require it.
    sourceAnyPortEnable: no

    # If the source is a RTSPS, RTMPS or HTTPS URL, and the source certificate is self-signed
    # or invalid, you can provide the fingerprint of the certificate in order to
    # validate it anyway. It can be obtained by running:
    # openssl s_client -connect source_ip:source_port </dev/null 2>/dev/null | sed -n '/BEGIN/,/END/p' > server.crt
    # openssl x509 -in server.crt -noout -fingerprint -sha256 | cut -d "=" -f2 | tr -d ':'
    sourceFingerprint:

    # If the source is an RTSP or RTMP URL, it will be pulled only when at least
    # one reader is connected, saving bandwidth.
    sourceOnDemand: no
    # If sourceOnDemand is "yes", readers will be put on hold until the source is
    # ready or until this amount of time has passed.
    sourceOnDemandStartTimeout: 10s
    # If sourceOnDemand is "yes", the source will be closed when there are no
    # readers connected and this amount of time has passed.
    sourceOnDemandCloseAfter: 10s

    # If the source is "redirect", this is the RTSP URL which clients will be
    # redirected to.
    sourceRedirect:

    # If the source is "publisher" and a client is publishing, do not allow another
    # client to disconnect the former and publish in its place.
    disablePublisherOverride: no

    # If the source is "publisher" and no one is publishing, redirect readers to this
    # path. It can be can be a relative path  (i.e. /otherstream) or an absolute RTSP URL.
    fallback:

    # If the source is "rpiCamera", these are the Raspberry Pi Camera parameters.
    # ID of the camera
    rpiCameraCamID: 0
    # width of frames (default 1920)
    rpiCameraWidth: 1920
    # height of frames (default 1080)
    rpiCameraHeight: 1080
    # flip horizontally (default false)
    rpiCameraHFlip: false
    # flip vertically (default false)
    rpiCameraVFlip: true
    # brightness [-1, 1]
    rpiCameraBrightness: 0
    # contrast [0, 16]
    rpiCameraContrast: 1
    # saturation [0, 16]
    rpiCameraSaturation: 1
    # sharpness [0, 16]
    rpiCameraSharpness: 1
    # exposure mode.
    # values: normal, short, long, custom
    rpiCameraExposure: normal
    # auto-white-balance mode.
    # values: auto, incandescent, tungsten, fluorescent, indoor, daylight, cloudy, custom
    rpiCameraAWB: auto
    # denoise operating mode.
    # values: auto, off, cdn_off, cdn_fast, cdn_hq
    rpiCameraDenoise: auto
    # fixed shutter speed, in microseconds.
    rpiCameraShutter: 0
    # metering mode of the AEC/AGC algorithm.
    # values: centre, spot, matrix, custom
    rpiCameraMetering: centre
    # fixed gain
    rpiCameraGain: 0
    # EV compensation of the image [-10, 10]
    rpiCameraEV: 0
    # Region of interest, in format x,y,width,height
    rpiCameraROI:
    # tuning file
    rpiCameraTuningFile:
    # sensor mode, in format [width]:[height]:[bit-depth]:[packing]
    # bit-depth and packing are optional.
    rpiCameraMode:
    # frames per second
    rpiCameraFPS: 30
    # period between IDR frames
    rpiCameraIDRPeriod: 60
    # bitrate
    rpiCameraBitrate: 1000000
    # H264 profile
    rpiCameraProfile: main
    # H264 level
    rpiCameraLevel: '4.1'

    # Username required to publish.
    # SHA256-hashed values can be inserted with the "sha256:" prefix.
    publishUser:
    # Password required to publish.
    # SHA256-hashed values can be inserted with the "sha256:" prefix.
    publishPass:
    # IPs or networks (x.x.x.x/24) allowed to publish.
    publishIPs: []

    # Username required to read.
    # SHA256-hashed values can be inserted with the "sha256:" prefix.
    readUser:
    # password required to read.
    # SHA256-hashed values can be inserted with the "sha256:" prefix.
    readPass:
    # IPs or networks (x.x.x.x/24) allowed to read.
    readIPs: []

    # Command to run when this path is initialized.
    # This can be used to publish a stream and keep it always opened.
    # This is terminated with SIGINT when the program closes.
    # The following environment variables are available:
    # * RTSP_PATH: path name
    # * RTSP_PORT: server port
    # * G1, G2, ...: regular expression groups, if path name is
    #   a regular expression.
    runOnInit:
    # Restart the command if it exits suddenly.
    runOnInitRestart: no

    # Command to run when this path is requested.
    # This can be used to publish a stream on demand.
    # This is terminated with SIGINT when the path is not requested anymore.
    # The following environment variables are available:
    # * RTSP_PATH: path name
    # * RTSP_PORT: server port
    # * G1, G2, ...: regular expression groups, if path name is
    #   a regular expression.
    runOnDemand:
    # Restart the command if it exits suddenly.
    runOnDemandRestart: no
    # Readers will be put on hold until the runOnDemand command starts publishing
    # or until this amount of time has passed.
    runOnDemandStartTimeout: 10s
    # The command will be closed when there are no
    # readers connected and this amount of time has passed.
    runOnDemandCloseAfter: 10s

    # Command to run when the stream is ready to be read, whether it is
    # published by a client or pulled from a server / camera.
    # This is terminated with SIGINT when the stream is not ready anymore.
    # The following environment variables are available:
    # * RTSP_PATH: path name
    # * RTSP_PORT: server port
    # * G1, G2, ...: regular expression groups, if path name is
    #   a regular expression.
    runOnReady:
    # Restart the command if it exits suddenly.
    runOnReadyRestart: no

    # Command to run when a clients starts reading.
    # This is terminated with SIGINT when a client stops reading.
    # The following environment variables are available:
    # * RTSP_PATH: path name
    # * RTSP_PORT: server port
    # * G1, G2, ...: regular expression groups, if path name is
    #   a regular expression.
    runOnRead:
    # Restart the command if it exits suddenly.
    runOnReadRestart: no
```

**Comment by aler9** [comment]
The configuration you posted doesn't have `source: rpiCamera` and therefore doesn't involve the camera stream. Please try this one:

```yml
paths:
  camera:
    source: rpiCamera
    rpiCameraHFlip: true
```

**Comment by yllekz** [comment]
Hmm ok so I stripped down my rtsp-simple-server.yml to only this:

```
paths:
  camera:
    source: rpiCamera
    rpiCameraVFlip: true
    rpiCameraHFlip: false
```

Making that adjustment seemed to take, but VFLip and HFlip are still not working for me.

As a side note, I also noticed that stopping and starting the containers (and often even totally rebooting) wasn't enough: Seems some process really likes to hang onto the rPi Camera. I can get it to come up every once in a while but it seems to be random. I have it working at the moment but it still isn't honoring the VFlip and HFlip settings.

```
12/09/2022 9:15:09 AM
[0:03:35.457355195] [114]  INFO Camera camera_manager.cpp:293 libcamera v0.0.1+54-d528119f
12/09/2022 9:15:09 AM
[0:03:35.537942863] [115]  INFO RPI raspberrypi.cpp:1414 Registered camera /base/soc/i2c0mux/i2c@1/ov5647@36 to Unicam device /dev/media0 and ISP device /dev/media1
12/09/2022 9:15:09 AM
[0:03:35.539000125] [114]  INFO Camera camera.cpp:839 Pipeline handler in use by another process
12/09/2022 9:15:09 AM
2022/12/09 14:15:09 INF [path camera] [rpicamera source] ERR: camera_create(): Camera.acquire() failed
12/09/2022 9:15:10 AM
2022/12/09 14:15:10 INF [HLS] [muxer cam] created (requested by 10.7.0.5)
12/09/2022 9:15:10 AM
2022/12/09 14:15:10 INF [HLS] [muxer cam] is converting into HLS, 1 track (H264)
12/09/2022 9:15:14 AM
[0:03:40.591638412] [118]  INFO Camera camera_manager.cpp:293 libcamera v0.0.1+54-d528119f
12/09/2022 9:15:14 AM
[0:03:40.674829675] [119]  INFO RPI raspberrypi.cpp:1414 Registered camera /base/soc/i2c0mux/i2c@1/ov5647@36 to Unicam device /dev/media0 and ISP device /dev/media1
12/09/2022 9:15:14 AM
[0:03:40.676510984] [118]  INFO Camera camera.cpp:839 Pipeline handler in use by another process
12/09/2022 9:15:14 AM
2022/12/09 14:15:14 INF [path camera] [rpicamera source] ERR: camera_create(): Camera.acquire() failed
12/09/2022 9:15:19 AM
[0:03:45.739876497] [122]  INFO Camera camera_manager.cpp:293 libcamera v0.0.1+54-d528119f
12/09/2022 9:15:19 AM
[0:03:45.826010433] [123]  INFO RPI raspberrypi.cpp:1414 Registered camera /base/soc/i2c0mux/i2c@1/ov5647@36 to Unicam device /dev/media0 and ISP device /dev/media1
12/09/2022 9:15:19 AM
[0:03:45.827081604] [122]  INFO Camera camera.cpp:839 Pipeline handler in use by another process
12/09/2022 9:15:19 AM
2022/12/09 14:15:19 INF [path camera] [rpicamera source] ERR: camera_create(): Camera.acquire() failed
12/09/2022 9:15:24 AM
[0:03:50.880475232] [126]  INFO Camera camera_manager.cpp:293 libcamera v0.0.1+54-d528119f
12/09/2022 9:15:24 AM
[0:03:50.965402863] [127]  INFO RPI raspberrypi.cpp:1414 Registered camera /base/soc/i2c0mux/i2c@1/ov5647@36 to Unicam device /dev/media0 and ISP device /dev/media1
12/09/2022 9:15:24 AM
[0:03:50.965985726] [126]  INFO Camera camera.cpp:839 Pipeline handler in use by another process
12/09/2022 9:15:24 AM
2022/12/09 14:15:24 INF [path camera] [rpicamera source] ERR: camera_create(): Camera.acquire() failed
12/09/2022 9:15:29 AM
[0:03:56.022348704] [130]  INFO Camera camera_manager.cpp:293 libcamera v0.0.1+54-d528119f
12/09/2022 9:15:29 AM
[0:03:56.111668693] [131]  INFO RPI raspberrypi.cpp:1414 Registered camera /base/soc/i2c0mux/i2c@1/ov5647@36 to Unicam device /dev/media0 and ISP device /dev/media1
12/09/2022 9:15:29 AM
[0:03:56.112330056] [130]  INFO Camera camera.cpp:839 Pipeline handler in use by another process
12/09/2022 9:15:29 AM
2022/12/09 14:15:29 INF [path camera] [rpicamera source] ERR: camera_create(): Camera.acquire() failed
12/09/2022 9:15:34 AM
[0:04:01.170800702] [134]  INFO Camera camera_manager.cpp:293 libcamera v0.0.1+54-d528119f
12/09/2022 9:15:34 AM
[0:04:01.251765196] [135]  INFO RPI raspberrypi.cpp:1414 Registered camera /base/soc/i2c0mux/i2c@1/ov5647@36 to Unicam device /dev/media0 and ISP device /dev/media1
12/09/2022 9:15:34 AM
[0:04:01.252380708] [134]  INFO Camera camera.cpp:839 Pipeline handler in use by another process
12/09/2022 9:15:34 AM
2022/12/09 14:15:34 INF [path camera] [rpicamera source] ERR: camera_create(): Camera.acquire() failed
12/09/2022 9:15:39 AM
[0:04:06.301350227] [138]  INFO Camera camera_manager.cpp:293 libcamera v0.0.1+54-d528119f
12/09/2022 9:15:40 AM
[0:04:06.382970027] [139]  INFO RPI raspberrypi.cpp:1414 Registered camera /base/soc/i2c0mux/i2c@1/ov5647@36 to Unicam device /dev/media0 and ISP device /dev/media1
12/09/2022 9:15:40 AM
[0:04:06.383797204] [138]  INFO Camera camera.cpp:839 Pipeline handler in use by another process
12/09/2022 9:15:40 AM
2022/12/09 14:15:40 INF [path camera] [rpicamera source] ERR: camera_create(): Camera.acquire() failed
12/09/2022 9:15:45 AM
[0:04:11.447383591] [142]  INFO Camera camera_manager.cpp:293 libcamera v0.0.1+54-d528119f
12/09/2022 9:15:45 AM
[0:04:11.538603559] [143]  INFO RPI raspberrypi.cpp:1414 Registered camera /base/soc/i2c0mux/i2c@1/ov5647@36 to Unicam device /dev/media0 and ISP device /dev/media1
12/09/2022 9:15:45 AM
[0:04:11.539497143] [142]  INFO Camera camera.cpp:839 Pipeline handler in use by another process
12/09/2022 9:15:45 AM
2022/12/09 14:15:45 INF [path camera] [rpicamera source] ERR: camera_create(): Camera.acquire() failed
12/09/2022 9:15:50 AM
[0:04:16.593693470] [146]  INFO Camera camera_manager.cpp:293 libcamera v0.0.1+54-d528119f
12/09/2022 9:15:50 AM
[0:04:16.675836112] [147]  INFO RPI raspberrypi.cpp:1414 Registered camera /base/soc/i2c0mux/i2c@1/ov5647@36 to Unicam device /dev/media0 and ISP device /dev/media1
12/09/2022 9:15:50 AM
[0:04:16.676602068] [146]  INFO Camera camera.cpp:839 Pipeline handler in use by another process
12/09/2022 9:15:50 AM
2022/12/09 14:15:50 INF [path camera] [rpicamera source] ERR: camera_create(): Camera.acquire() failed
12/09/2022 9:15:55 AM
[0:04:21.729150482] [150]  INFO Camera camera_manager.cpp:293 libcamera v0.0.1+54-d528119f
12/09/2022 9:15:55 AM
[0:04:21.810085092] [151]  INFO RPI raspberrypi.cpp:1414 Registered camera /base/soc/i2c0mux/i2c@1/ov5647@36 to Unicam device /dev/media0 and ISP device /dev/media1
12/09/2022 9:15:55 AM
[0:04:21.810660235] [150]  INFO Camera camera.cpp:839 Pipeline handler in use by another process
12/09/2022 9:15:55 AM
2022/12/09 14:15:55 INF [path camera] [rpicamera source] ERR: camera_create(): Camera.acquire() failed
12/09/2022 9:16:00 AM
[0:04:26.861702535] [154]  INFO Camera camera_manager.cpp:293 libcamera v0.0.1+54-d528119f
12/09/2022 9:16:00 AM
[0:04:26.942851484] [155]  INFO RPI raspberrypi.cpp:1414 Registered camera /base/soc/i2c0mux/i2c@1/ov5647@36 to Unicam device /dev/media0 and ISP device /dev/media1
12/09/2022 9:16:00 AM
[0:04:26.943755865] [154]  INFO Camera camera.cpp:839 Pipeline handler in use by another process
12/09/2022 9:16:00 AM
2022/12/09 14:16:00 INF [path camera] [rpicamera source] ERR: camera_create(): Camera.acquire() failed
12/09/2022 9:16:05 AM
[0:04:31.992272506] [158]  INFO Camera camera_manager.cpp:293 libcamera v0.0.1+54-d528119f
12/09/2022 9:16:05 AM
[0:04:32.069828245] [159]  INFO RPI raspberrypi.cpp:1414 Registered camera /base/soc/i2c0mux/i2c@1/ov5647@36 to Unicam device /dev/media0 and ISP device /dev/media1
12/09/2022 9:16:05 AM
[0:04:32.070487629] [158]  INFO Camera camera.cpp:839 Pipeline handler in use by another process
12/09/2022 9:16:05 AM
2022/12/09 14:16:05 INF [path camera] [rpicamera source] ERR: camera_create(): Camera.acquire() failed
12/09/2022 9:16:10 AM
[0:04:37.121417651] [162]  INFO Camera camera_manager.cpp:293 libcamera v0.0.1+54-d528119f
12/09/2022 9:16:10 AM
[0:04:37.201930091] [163]  INFO RPI raspberrypi.cpp:1414 Registered camera /base/soc/i2c0mux/i2c@1/ov5647@36 to Unicam device /dev/media0 and ISP device /dev/media1
12/09/2022 9:16:10 AM
[0:04:37.202589901] [162]  INFO Camera camera.cpp:839 Pipeline handler in use by another process
12/09/2022 9:16:10 AM
2022/12/09 14:16:10 INF [path camera] [rpicamera source] ERR: camera_create(): Camera.acquire() failed
12/09/2022 9:16:15 AM
[0:04:42.252909684] [166]  INFO Camera camera_manager.cpp:293 libcamera v0.0.1+54-d528119f
12/09/2022 9:16:15 AM
[0:04:42.339208487] [167]  INFO RPI raspberrypi.cpp:1414 Registered camera /base/soc/i2c0mux/i2c@1/ov5647@36 to Unicam device /dev/media0 and ISP device /dev/media1
12/09/2022 9:16:15 AM
[0:04:42.340112758] [166]  INFO Camera camera.cpp:839 Pipeline handler in use by another process
12/09/2022 9:16:15 AM
2022/12/09 14:16:15 INF [path camera] [rpicamera source] ERR: camera_create(): Camera.acquire() failed
12/09/2022 9:16:21 AM
[0:04:47.389277564] [170]  INFO Camera camera_manager.cpp:293 libcamera v0.0.1+54-d528119f
12/09/2022 9:16:21 AM
[0:04:47.469380688] [171]  INFO RPI raspberrypi.cpp:1414 Registered camera /base/soc/i2c0mux/i2c@1/ov5647@36 to Unicam device /dev/media0 and ISP device /dev/media1
12/09/2022 9:16:21 AM
[0:04:47.469809074] [170]  INFO Camera camera.cpp:839 Pipeline handler in use by another process
12/09/2022 9:16:21 AM
2022/12/09 14:16:21 INF [path camera] [rpicamera source] ERR: camera_create(): Camera.acquire() failed
12/09/2022 9:16:26 AM
[0:04:52.518378531] [174]  INFO Camera camera_manager.cpp:293 libcamera v0.0.1+54-d528119f
12/09/2022 9:16:26 AM
[0:04:52.605032380] [175]  INFO RPI raspberrypi.cpp:1414 Registered camera /base/soc/i2c0mux/i2c@1/ov5647@36 to Unicam device /dev/media0 and ISP device /dev/media1
12/09/2022 9:16:26 AM
[0:04:52.605703357] [174]  INFO Camera camera.cpp:839 Pipeline handler in use by another process
12/09/2022 9:16:26 AM
2022/12/09 14:16:26 INF [path camera] [rpicamera source] ERR: camera_create(): Camera.acquire() failed
12/09/2022 9:16:31 AM
[0:04:57.655983475] [178]  INFO Camera camera_manager.cpp:293 libcamera v0.0.1+54-d528119f
12/09/2022 9:16:31 AM
[0:04:57.741842895] [179]  INFO RPI raspberrypi.cpp:1414 Registered camera /base/soc/i2c0mux/i2c@1/ov5647@36 to Unicam device /dev/media0 and ISP device /dev/media1
12/09/2022 9:16:31 AM
[0:04:57.742311318] [178]  INFO Camera camera.cpp:839 Pipeline handler in use by another process
12/09/2022 9:16:31 AM
2022/12/09 14:16:31 INF [path camera] [rpicamera source] ERR: camera_create(): Camera.acquire() failed
12/09/2022 9:16:36 AM
[0:05:02.791102459] [182]  INFO Camera camera_manager.cpp:293 libcamera v0.0.1+54-d528119f
12/09/2022 9:16:36 AM
[0:05:02.874874587] [183]  INFO RPI raspberrypi.cpp:1414 Registered camera /base/soc/i2c0mux/i2c@1/ov5647@36 to Unicam device /dev/media0 and ISP device /dev/media1
12/09/2022 9:16:36 AM
[0:05:02.875884803] [182]  INFO Camera camera.cpp:839 Pipeline handler in use by another process
12/09/2022 9:16:36 AM
2022/12/09 14:16:36 INF [path camera] [rpicamera source] ERR: camera_create(): Camera.acquire() failed
12/09/2022 9:16:41 AM
[0:05:07.926645170] [186]  INFO Camera camera_manager.cpp:293 libcamera v0.0.1+54-d528119f
12/09/2022 9:16:41 AM
[0:05:08.011210227] [187]  INFO RPI raspberrypi.cpp:1414 Registered camera /base/soc/i2c0mux/i2c@1/ov5647@36 to Unicam device /dev/media0 and ISP device /dev/media1
12/09/2022 9:16:41 AM
[0:05:08.011859778] [186]  INFO Camera camera.cpp:839 Pipeline handler in use by another process
12/09/2022 9:16:41 AM
2022/12/09 14:16:41 INF [path camera] [rpicamera source] ERR: camera_create(): Camera.acquire() failed
12/09/2022 9:16:46 AM
[0:05:13.061336004] [190]  INFO Camera camera_manager.cpp:293 libcamera v0.0.1+54-d528119f
12/09/2022 9:16:46 AM
[0:05:13.154955268] [191]  INFO RPI raspberrypi.cpp:1414 Registered camera /base/soc/i2c0mux/i2c@1/ov5647@36 to Unicam device /dev/media0 and ISP device /dev/media1
12/09/2022 9:16:46 AM
[0:05:13.156152890] [190]  INFO Camera camera.cpp:839 Pipeline handler in use by another process
12/09/2022 9:16:46 AM
2022/12/09 14:16:46 INF [path camera] [rpicamera source] ERR: camera_create(): Camera.acquire() failed
12/09/2022 9:16:51 AM
[0:05:18.207019468] [194]  INFO Camera camera_manager.cpp:293 libcamera v0.0.1+54-d528119f
12/09/2022 9:16:51 AM
[0:05:18.293117582] [195]  INFO RPI raspberrypi.cpp:1414 Registered camera /base/soc/i2c0mux/i2c@1/ov5647@36 to Unicam device /dev/media0 and ISP device /dev/media1
12/09/2022 9:16:51 AM
[0:05:18.294032687] [194]  INFO Camera camera.cpp:839 Pipeline handler in use by another process
12/09/2022 9:16:51 AM
2022/12/09 14:16:51 INF [path camera] [rpicamera source] ERR: camera_create(): Camera.acquire() failed
12/09/2022 9:16:56 AM
[0:05:23.344665813] [198]  INFO Camera camera_manager.cpp:293 libcamera v0.0.1+54-d528119f
12/09/2022 9:16:57 AM
[0:05:23.424759342] [199]  INFO RPI raspberrypi.cpp:1414 Registered camera /base/soc/i2c0mux/i2c@1/ov5647@36 to Unicam device /dev/media0 and ISP device /dev/media1
12/09/2022 9:16:57 AM
[0:05:23.425632077] [198]  INFO Camera camera.cpp:839 Pipeline handler in use by another process
12/09/2022 9:16:57 AM
2022/12/09 14:16:57 INF [path camera] [rpicamera source] ERR: camera_create(): Camera.acquire() failed
12/09/2022 9:17:02 AM
[0:05:28.479518387] [202]  INFO Camera camera_manager.cpp:293 libcamera v0.0.1+54-d528119f
12/09/2022 9:17:02 AM
[0:05:28.565397735] [203]  INFO RPI raspberrypi.cpp:1414 Registered camera /base/soc/i2c0mux/i2c@1/ov5647@36 to Unicam device /dev/media0 and ISP device /dev/media1
12/09/2022 9:17:02 AM
[0:05:28.566133693] [202]  INFO Camera camera.cpp:839 Pipeline handler in use by another process
12/09/2022 9:17:02 AM
2022/12/09 14:17:02 INF [path camera] [rpicamera source] ERR: camera_create(): Camera.acquire() failed
12/09/2022 9:17:07 AM
[0:05:33.619373784] [206]  INFO Camera camera_manager.cpp:293 libcamera v0.0.1+54-d528119f
12/09/2022 9:17:07 AM
[0:05:33.709566126] [207]  INFO RPI raspberrypi.cpp:1414 Registered camera /base/soc/i2c0mux/i2c@1/ov5647@36 to Unicam device /dev/media0 and ISP device /dev/media1
12/09/2022 9:17:07 AM
[0:05:33.710067197] [206]  INFO Camera camera.cpp:839 Pipeline handler in use by another process
12/09/2022 9:17:07 AM
2022/12/09 14:17:07 INF [path camera] [rpicamera source] ERR: camera_create(): Camera.acquire() failed
12/09/2022 9:17:12 AM
[0:05:38.759679017] [210]  INFO Camera camera_manager.cpp:293 libcamera v0.0.1+54-d528119f
12/09/2022 9:17:12 AM
[0:05:38.879829435] [211]  INFO RPI raspberrypi.cpp:1414 Registered camera /base/soc/i2c0mux/i2c@1/ov5647@36 to Unicam device /dev/media0 and ISP device /dev/media1
12/09/2022 9:17:12 AM
[0:05:38.880705652] [210]  INFO Camera camera.cpp:839 Pipeline handler in use by another process
12/09/2022 9:17:12 AM
2022/12/09 14:17:12 INF [path camera] [rpicamera source] ERR: camera_create(): Camera.acquire() failed
```

**Comment by yllekz** [comment]
I'll have to dig into this further. If I remove the ``RTSP_PATHS_CAM_SOURCE=rpiCamera`` then the container errors out even more and completely fails to start. Seems it's needed.

**Comment by yllekz** [comment]
Yes I understand that part.

I got sidelined by another issue that cropped up: libcamera was suddenly not detected. Took me a bit to figure it out but I swapped over to the ``latest-rpi`` tag from ``latest`` and that cleared up the errors.

All good now. Thanks for letting me know the names of the env vars for the flip attributes so I can just inject them straight into the docker-compose. I'm not a fan of a secondary config file for this stuff so this is a nice simple way to get it up and running. Thanks!

**Comment by aler9** [comment]
libcamera is constantly being updated in a backward-incompatible way by Raspberry Pi maintainers, and that's the reason why sometimes it's not recognized. Therefore we found a way to ship it inside a dedicated image (`latest-rpi`) and in this way it should always remain available.
